src/block/block_addr.c

```diff
@@ -36,8 +39,23 @@ __block_addr_unpack(WT_BLOCK *block, const uint8_t **pp, size_t addr_size, uint3
     WT_RET(__wt_vunpack_uint(pp, 0, &s));
     WT_RET(__wt_vunpack_uint(pp, 0, &c));
     i = 0;
-    if (addr_size != 0 && WT_PTRDIFF(*pp, begin) < addr_size)
-        WT_RET(__wt_vunpack_uint(pp, 0, &i));
+    flags = 0;
+    if (addr_size != 0 && WT_PTRDIFF(*pp, begin) < addr_size) {
+        flags = **pp;
+        ++(*pp);
+        if (LF_ISSET(WT_BLOCK_COOKIE_FILEID))
+            WT_RET(__wt_vunpack_uint(pp, 0, &i));
+    }
+
+    /*
+     * If there's an address cookie size and either there is no flag value or the flag value is the
+     * object ID flag by itself, assert the cookie was entirely consumed. Future extensions will use
+     * different cookie flag values (although the file-ID flag might still be set, our test is for
+     * equality).
+     */
+    WT_ASSERT(session,
+      addr_size == 0 || (flags != 0 && flags != WT_BLOCK_COOKIE_FILEID) ||
+        WT_PTRDIFF(*pp, begin) == addr_size);
```

Comment:
Can we strengthen this assert?  Like:
```
    WT_ASSERT(session,
      addr_size == 0 || ((flags != 0 && flags != WT_BLOCK_COOKIE_FILEID) &&
        WT_PTRDIFF(*pp, begin) == addr_size));
```

Reply:
@ddanderson, this doesn't look right to me.

The current assert reads "if we got an address size, then if there weren't any flags or the flags were flags this release knows about, then we should have read all of the address cookie".

I'd read your assert as "if we got an address size, then there must be flags, and they may not be 0 or the flags we know about, and we must have read all the address cookie"?

I think there's an extra set of parenthesis in there, too, so maybe that's not what you intended to type?

Or maybe I'm just misreading it!!